backend/httpclient/http_client.go

```diff
@@ -211,6 +211,7 @@ func DefaultMiddlewares() []Middleware {
 		CustomHeadersMiddleware(),
 		ContextualMiddleware(),
 		ErrorSourceMiddleware(),
+		DataSourceMetricsMiddleware(),
```

Comment:
I would suggest putting this after the tracing middleware so that it starts recording the timing metrics as early as possible.

Reply:
Done